API/projectClass/projectClass.m

```diff
@@ -676,84 +665,81 @@
             % 
             % problem.addContrast('contrast 1', 'nba', 'Silicon');
             allowedNames = obj.getAllAllowedNames();
-            obj.contrasts.addContrast(allowedNames,varargin);
+            obj.contrasts.addContrast(allowedNames, varargin{:});
             
         end
 
-        function obj = removeContrast(obj,varargin)
+        function obj = removeContrast(obj, contrastInput)
             % Removes a specified contrast parameter. Expects
             % index or name of resolution to remove
             %
             % problem.removeContrast(1);
-            obj.contrasts.removeContrast(varargin{:});
+            obj.contrasts.removeContrast(contrastInput);
         end
 
         
-        function obj = setContrast(obj,varargin)   
+        function obj = setContrast(obj, contrastInput, varargin)   
             % Allow setting of all parameters in terms of name value pairs.
             % First input must be contrast number or name, subsequent
             % inputs are name / value pairs for the parts involved
             %
             % problem.setContrast(1, 'name', 'contrast')
-            contrastInput = varargin{1};
-            inputVals = varargin(2:end);
                         
             % Get the list of allowed values depending on what is
             % set for the other contrasts.
             allowedValues = obj.getAllAllowedNames;
             
             % Call the setContrast method
-            obj.contrasts.setContrast(contrastInput,allowedValues,inputVals);
+            obj.contrasts.setContrast(contrastInput, allowedValues, varargin{:});
             
         end
         
-        function obj = setContrastModel(obj,varargin)
-            % Edits the model of an existing contrast parameter. Expects the
-            % index of contrast parameter and cell array of layer names
+        function obj = setContrastModel(obj, contrastInput, model)
+            % Edits the model of an existing contrast parameter. Expects
+            % the index of contrast parameter and cell array of layer names
             %
             % problem.setContrastModel(1, {'layer 1'})
-            contrastInput = varargin{1};
-            inputVals = varargin(2:end);
                         
             % Make a different allowed list depending on whether 
             % it is custom or layers
             if ~strcmpi(obj.modelType, {'custom layers','custom xy'})
                 % Standard Layers
                 allowedValues = obj.layers.getLayersNames();
-                inputVals = inputVals{:};
             else
                 % Custom models
                 allowedValues = obj.customFile.getCustomNames();
             end
             
             % Call the setContrastModel method
-            obj.contrasts.setContrastModel(contrastInput, obj.modelType, allowedValues, inputVals);
+            obj.contrasts.setContrastModel(contrastInput, obj.modelType, allowedValues, model);
```

Comment:
Did we fix the issue where model could be a string or a cell?

Reply:
No, because it depends whether or not we have a standard layers model or a custom model. The routine is set to cover all eventualities properly.